src/net.h

```diff
@@ -769,11 +769,28 @@ class CNode
 class NetEventsInterface
 {
 public:
-    virtual bool ProcessMessages(CNode* pnode, std::atomic<bool>& interrupt) = 0;
-    virtual bool SendMessages(CNode* pnode) EXCLUSIVE_LOCKS_REQUIRED(pnode->cs_sendProcessing) = 0;
+    /** Initialize a peer (setup state, queue any initial messages) */
```

Comment:
I don't think it's a big deal, but if you're going to add these comments to the interface class, I think there are some potential improvements:

- "queue any initial messages" is only true for the version message for outbound peers. Since it's not universally true, it probably doesn't belong in the description of the interface.
- `SendMessages()` return value is not used to indicate if more work is done. The function always returns `true` and that value is discarded by the caller. A future commit could change the function to return `void`.
- `ProcessMessages()` _does_ return a bool that indicates whether there's more work.

Eventually, these functions could all be updated to just take a single `CNode&` argument, which would be a very clean interface (#20228 removes the `update_connection_time` arg from `FinalizeNode(), and I have another branch that eliminates `interrupt`).

Reply:
`SendMessages` lost its new lock annotation in the rebase, so I've added the doc from `ProcessMessages` return value.

I think "any initial messages" covers nothing, just a version message, or anything else we might think up fine. The return value of `SendMessages` is "used" in the unit tests, so haven't removed that in this PR. Looking at the `CNode&` thing in #20758